tracing-core/src/subscriber.rs

```diff
@@ -314,6 +314,23 @@ pub trait Subscriber: 'static {
         let _ = id;
     }
 
+    /// Returns a type representing this subscriber's view of the current span.
+    ///
+    /// If subscribers track a current span, they should override this function
+    /// to return [`Current::new`] if the thread from which this method is
+    /// called is inside a span, or [`Current::none`] if the thread is not
+    /// inside a span.
+    ///
+    /// By default, this returns a value indicating that the subscriber
+    /// does **not** track what span is current. If the subscriber does not
+    /// implement a current span, it should not override this method.
+    ///
+    /// [`Current::new`]: ../span/struct.Current.html#tymethod.new
+    /// [`Current::none`]: ../span/struct.Current.html#tymethod.none
+    fn current_span(&self) -> span::Current {
```

Comment:
We should also add impls for this function for the existing tracing subscribers.

Reply:
Yeah, I thought I'd hold off on that until `tracing-core` 0.1.1 is published to crates.io, as it would require switching everything back to path dependencies. Since most of the existing subscribers are unreleased, I thought it was important not to break git dependencies on those crates, as everyone using them will necessarily be depending on them from git.